### ISSUE TITLE
Touch on touch-panel is not working within the application

### ISSUE THREAD

**Issue by luisaodh** [fyne-io/fyne#2876]
I want to build a fyne app for a smart home usecase on a debian bullseye virtual machine. The target device is an embedded device with a connected touch panel with linux debian operating system and arm architecture.

### Describe the bug:

I use fyne-cross to compile the code. Although the touch does work on the system and thus on other applications, the problem is that the touch does not work in the fyne application. The target device uses wayland-weston. In addition, the application only starts when a mouse is plugged in. The same problem does not occur on an android device. Maybe the problem is that for linux/arm compilations the desktop driver gets choosen and there is no touch driver like there is for android devices. Do you have any ideas to fix this problem to enable the touch?

### To Reproduce:
I use the following command to cross-compile:
```fyne-cross linux -arch arm -tags wayland,gles,mesa```

### Device:
 - **OS:** Linux
 - **Version:** Debian GNU/Linux 11 (bullseye)
 - **Go version:** go1.18 linux/amd64
 - **Fyne version:** v2.1.3
 - **Fyne-cross version:** v1.1.2

**Comment by Jacalz** [comment]
Does it work without the `wayland` tag?

**Comment by Bluebugs** [comment]
I don't know if Fyne has any code to do this on its own, but currently glfw doesn't have proper touch input as far as I know: https://github.com/glfw/glfw/issues/42 .

**Comment by Jacalz** [comment]
I've been able to use my touch screen with Fyne apps as far as I know. That has however been on on X11/Wayland but without the `wayland` tag.

**Comment by luisaodh** [comment]
Thank you for all you suggestions!

Without the ```wayland``` tag I get the error ```PlatformError: X11: The DISPLAY environment variable is missing
panic: NotInitialized: The GLFW library is not initialized``` when running the app.

With the ```wayland``` and ```mobile``` tag, I can not cross-compile since I get the following error output:
[i] Target: linux/arm
[i] Cleaning target directories...
[✓] "bin" dir cleaned: /home/devel/Dokumente/smart-home/fyne-cross/bin/linux-arm
[✓] "dist" dir cleaned: /home/devel/Dokumente/smart-home/fyne-cross/dist/linux-arm
[✓] "temp" dir cleaned: /home/devel/Dokumente/smart-home/fyne-cross/tmp/linux-arm
[i] Checking for go.mod: /home/devel/Dokumente/smart-home/go.mod
[✓] go.mod found
[i] Building binary...
\# fyne.io/fyne/v2/internal/driver/mobile
/go/pkg/mod/fyne.io/fyne/v2@v2.1.3/internal/driver/mobile/driver.go:333:32: undefined: tapYOffset
/go/pkg/mod/fyne.io/fyne/v2@v2.1.3/internal/driver/mobile/driver.go:341:32: undefined: tapYOffset
/go/pkg/mod/fyne.io/fyne/v2@v2.1.3/internal/driver/mobile/driver.go:351:32: undefined: tapYOffset
[✗] exit status 2

And with just the ```mobile``` tag, I get the error ```XOpenDisplay failed``` when running the app.

Do you have any ideas how to fix one of these problems?

**Comment by Jacalz** [comment]
Thanks for testing. Would you mind adding a minimal code example (or alternatively testing if you see the same things with fyne_demo)? I think the bug issues template should contain some more information about that.

What window manager are you using? Does it, by any chance, not support X11?

Also, I see that you mentioned a “mesa” tag in your initial information. Where did you find out about that tag? I have never seen it been used before and it’s not one we define here in Fyne as far as I know.

That compile error is a bit strange. Looks like something is wrong with our build tag handling when using Wayland and mobile.

**Comment by luisaodh** [comment]
I can send you a Dropbox link via email so you can view my code. If that is okay for you, could you please send me your email address? I can not provide you with the link to the git repository, because it is an internal repository of the company.

My window manager is wayland. As far as I know, it does not support X11.

I tried the cross-compilation without the ```mesa``` tag, but get additional error messages.
Compiling with the command ```fyne-cross linux -arch arm -tags gles,wayland -image fyne-cross:v1.1-linux-arm-luisa``` and running the program on my target device results in the following error message: (I use an custom image because I need an additional dependency, see issue https://github.com/fyne-io/fyne-cross/issues/95)
```
PlatformError: Wayland: Standard cursor "crosshair" not found
PlatformError: Wayland: Standard cursor "hand2" not found
PlatformError: Wayland: Standard cursor "sb_h_double_arrow" not found
PlatformError: Wayland: Standard cursor "sb_v_double_arrow" not found
PlatformError: Wayland: Focusing a window requires user interaction
PlatformError: Wayland: Window position retrieval not supported
GLFW: An uncaught error has occured: PlatformError: Wayland: Standard cursor not found
GLFW: Please report this bug in the Go package immediately.
PlatformError: Wayland: Standard cursor not found
```

To compare it: The two error messages with GLFW are new compared to compiling it with the ```mesa``` tag
Compiling with the command ```fyne-cross linux -arch arm -tags gles,mesa,wayland -image fyne-cross:v1.1-linux-arm-luisa``` and running the program on my target device results in the following error message:
```
PlatformError: Wayland: Standard cursor "crosshair" not found
PlatformError: Wayland: Standard cursor "hand2" not found
PlatformError: Wayland: Standard cursor "sb_h_double_arrow" not found
PlatformError: Wayland: Standard cursor "sb_v_double_arrow" not found
PlatformError: Wayland: Focusing a window requires user interaction
PlatformError: Wayland: Window position retrieval not supported
PlatformError: Wayland: Standard cursor not found
```

**Comment by Jacalz** [comment]
> It is the new replacement for X11, but it can support old apps through "XWayland" which I guess is what Jacob was indicating you could test..

Indeed. As you can run Wayland without support for XWayland, there is a possibility that the window manager in question doesn’t support XWayland (that’s not very common but not impossible).

> I can send you a Dropbox link via email so you can view my code. If that is okay for you, could you please send me your email address? I can not provide you with the link to the git repository, because it is an internal repository of the company.

Thanks for the offer, but it would probably be better if you could just try to replicate with something like fyne_demo (see cmd/fyne_demo: https://github.com/fyne-io/fyne/tree/master/cmd/fyne_demo) instead.

Where did you find information about that “mesa” tag? It’s not one that we are aware of as far as I know.

**Comment by luisaodh** [comment]
I don't remember where I found the ```mesa``` tag. I am really sorry, also google did not help finding something.

I tried the fyne-demo with and without the ```mesa``` tag and both results in the same error message:
```
PlatformError: Wayland: Standard cursor "crosshair" not found
PlatformError: Wayland: Standard cursor "hand2" not found
PlatformError: Wayland: Standard cursor "sb_h_double_arrow" not found
PlatformError: Wayland: Standard cursor "sb_v_double_arrow" not found
PlatformError: Wayland: Focusing a window requires user interaction
PlatformError: Wayland: Setting window icon not supported
PlatformError: Wayland: Window position retrieval not supported
GLFW: An uncaught error has occured: PlatformError: Wayland: Standard cursor not found
GLFW: Please report this bug in the Go package immediately.
PlatformError: Wayland: Standard cursor not found
```
The touch also does not work within the fyne-demo application. And I used the standard image and not a custom image.
So I guess I can just omit the ```mesa``` tag.

I don't know if the Yocto Image I use has XWayland, but I try to get one with XWayland and then try it out.

**Comment by luisaodh** [comment]
I checked and my Yocto Image has XWayland. When I compile without wayland I get the earlier mentioned error:
```PlatformError: X11: The DISPLAY environment variable is missing panic: NotInitialized: The GLFW library is not initialized```
But could it be that it needs both the tags ```wayland``` and ```mobile``` and something is wrong there?

**Comment by luisaodh** [comment]
I managed to enable the touch by changing the weston.ini file on my Yocto image to enable xwayland. Then I cross-compiled the program without the ```wayland``` tag. The weston.ini can be found here: /etc/xdg/weston/weston.ini
In the core section I added ```xwayland=true```.

**Comment by andydotxyz** [comment]
Great to hear it is working, but I don't think this should be closed. Wayland support is being worked on and having touch panels not work is a problem.

**Comment by luisaodh** [comment]
It is a touch panel from DH-electronics. I am using a [PDK 2](https://www.dh-electronics.com/produkte/dhsom-system-on-modules/dh-pdk2) with a development Touch Panel. But this Touch Panel is also build in the [DHMI 7](https://www.dh-electronics.com/produkte/dhmi-touch-panel-computer/dhmi-7). And I am using [DHCOM i.MX6x](https://www.dh-electronics.com/produkte/dhsom-system-on-modules/dhcom-imx6x) in the DualLite Version. If you have any questions or problems regarding the Touch Panel, you can write swellnitz@dh-electronics.com. For ordering products, please contact info@dh-electronics.com.
